tealiumlibrary/src/main/java/com/tealium/core/collection/ConnectivityCollector.kt

```diff
@@ -26,17 +26,26 @@ class ConnectivityCollector(context: Context, private val connectivityRetriever:
     private val telephonyManager =
         context.applicationContext.getSystemService(Service.TELEPHONY_SERVICE) as TelephonyManager
 
-    override val carrier: String = telephonyManager.networkOperatorName ?: ""
-    override val carrierIso: String = telephonyManager.networkCountryIso ?: ""
-    override val carrierMcc: String =
-        if (telephonyManager.networkOperator.length > 3) telephonyManager.networkOperator.substring(
-            0,
-            3
-        ) else ""
-    override val carrierMnc: String =
-        if (telephonyManager.networkOperator.length > 3) telephonyManager.networkOperator.substring(
-            3
-        ) else ""
+    override val carrier: String
+        get() {
+            return telephonyManager.networkOperatorName ?: ""
+        }
+    override val carrierIso: String
+        get() {
+            return telephonyManager.networkCountryIso ?: ""
+        }
+    override val carrierMcc: String
+        get() {
+            return if (telephonyManager.networkOperator.isNotBlank() && telephonyManager.networkOperator.length > 3)
```

Comment:
it would be better to extract the `networkOperator` either through a local variable or a `let` so that it cannot have changed between the if checks and the subsequent `substring`
e.g.
```
val operator = telephonyManager.networkOperator
return if (operator.isNotBlank() && operator.length > 3)
    operator.substring(0, 3)
else ""
```

or using `let`
```
return telephonyManager.networkOperator.let { operator ->
    if (operator.isNotBlank() && operator.length > 3)
        operator.substring(0, 3)
    else ""
}
```